cli/script/command-executor.ts

```diff
@@ -532,52 +515,56 @@ function loginWithAccessTokenInternal(serverUrl: string): Promise<void> {
                 return;
             }
 
-            if (!accessToken) {
+            var connectionInfo: ILegacyLoginConnectionInfo|ILoginConnectionInfo;
+            try {
+                var decoded: string = base64.decode(accessToken);
+                connectionInfo = JSON.parse(decoded);
+            } catch (ex) {
+            }
```

Comment:
With existing helper methods, you can simplify lines 518-523 to:

    var decoded: string = tryBase64Decode(accessToken);
    var connectionInfo: ILegacyLoginConnectionInfo|ILoginConnectionInfo = tryJSON(decoded);